### DISCUSSION TITLE
Error tracking

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#6619] [Feature Requests]
What is the recommended way (if there is one) to track unexpected programming errors (status >= 500) in Directus using tools like Sentry?

**Comment by rijkvanzanten** [comment]
There's hooks at different points of the server startup (https://docs.directus.io/guides/api-hooks/#event-format-options) like `init` / `routes.init`, and `middlewares.init` you can use to plugin custom code / middleware like Sentry 👍🏻

**Reply by nickluger** [reply]
OK, perfect thank you!

**Reply by t7tran** [reply]
Do you have any suggestions for the browser app itself? It would be useful to track errors introduced in our custom code/modules. Thanks.

**Reply by joselcvarela** [reply]
Hello @t7tran 
I don't think there's a standard way to do so. Though, you can inject your third party on a custom interface. For example (not tested):

```ts
import InterfaceComponent from './interface.vue';
import Sentry from 'sentry';

const sentry = new Sentry(window)

export default {
	id: 'custom',
	name: 'Custom',
	description: 'This is my custom interface!',
	icon: 'box',
	component: InterfaceComponent,
	types: ['string'],
};
```

**Reply by rijkvanzanten** [reply]
True! Though that would register Sentry every time the interface is registered 🤔 

That being said, a workaround for modules/interfaces could be to do that registration in the onMount and onUnmounted hooks in Vue itself

**Reply by nickluger** [reply]
Plus: check a global flag like `window.hasRegisteredSentry` in your `mounted/onMount` to make sure sentry is registered only once.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
